### PR TITLE
feat: add change userdata logic

### PR DESCRIPTION
**Pull request by Iliustrator**

### CODE REVIEW

File: src/components/UserProfile/UserProfile.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            encodeImageFileAsURL(e);
+        }
+    }
+    const encodeImageFileAsURL = (element) => {
+        const file = element.target.files[0];
+        const reader = new FileReader();
+        reader.readAsDataURL(file);
+        reader.onload = () => {
+            encodedImage.name = encodeURIComponent(element.target.files[0].name);
+            trasformString(reader.result);
```
Comment (Pr0bka) [inline src/components/UserProfile/UserProfile.js:29]: maybe transformString?

File: src/components/UserProfile/UserProfile.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+
+    const trasformString = (string) => {
```
Comment (Pr0bka) [inline src/components/UserProfile/UserProfile.js:33]: maybe transformString?

File: src/components/UserProfile/UserProfile.module.css [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    border-color: #aa0b20;
+    background-color: #ffeff1;
+  }
+
+  .formContainer {
+    display: flex;
+    justify-content: center;
+  }
+
+  .currencySelector {
```
Comment (Pr0bka) [inline src/components/UserProfile/UserProfile.module.css:73]: Do not being used in js file

File: src/components/UserProfile/UserProfile.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
+import useHomeApi from "../../hooks/useHomeApi";
+import classes from './UserProfile.module.css';
+import { getDownloadURL, storage, ref } from '../../services/firebase';
+
+const UserProfile = () => {
+    const { putUser, postUserAvatar } = useHomeApi();
+    const userData = {};
+    const encodedImage = {};
```
Comment (AntonOvc) [inline src/components/UserProfile/UserProfile.js:8]: pls specify fields which would be added to the object at least as undefined

**Review by Pr0bka — CHANGES_REQUESTED** [review]

File: src/components/UserProfile/UserProfile.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            postUserAvatar(encodedImage.code, encodedImage.name)
+                .then(((e) => {
+                    getDownloadURL(ref(storage, e.path))
+                        .then((e) => {
+                            console.log(e);
+                            userData.avatar = e;
+                            putUser(JSON.stringify(userData));
+                        })
+                }));
+        } else if (userData.name) {
```
Comment (AntonOvc) [inline src/components/UserProfile/UserProfile.js:50]: postUserAvatar(encodedImage.code, encodedImage.name)
                 .then((e) => {
                    return getDownloadURL(ref(storage, e.path))
                 })
                   .then((e) => {
                             console.log(e);
                             userData.avatar = e;
                             putUser(JSON.stringify(userData));
                         })

Comment (AntonOvc) [reply src/components/UserProfile/UserProfile.js:50]: and add catch with console warn

File: src/components/UserProfile/UserProfile.js [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+            putUser(JSON.stringify(userData));
+        } else {
```
Comment (AntonOvc) [inline src/components/UserProfile/UserProfile.js:52]: else return is redundant

**Review by AntonOvc — CHANGES_REQUESTED** [review]
looks good just a few comments

**Review by AntonOvc — APPROVED** [review]

**Review by Pr0bka — APPROVED** [review]

**Pull request merged**